language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/RegexpCompletionProvider.java

```diff
@@ -0,0 +1,688 @@
+        } else if (nodeAtCursor.kind() == SyntaxKind.RE_BRACED_QUANTIFIER
+                && nodeAtCursor.textRange().endOffset() == ctx.getCursorPositionInTree()) {
+            completionItems.add(new SnippetCompletionItem(ctx, new SnippetBlock("?", "?", "?",
+                    ItemResolverConstants.VALUE_TYPE, SnippetBlock.Kind.VALUE)));
+        } else if (isCursorAfterReFlag(nodeAtCursor)) {
+            // Eg: re `(?i<cursor>)`, re `(?i-m<cursor>)`
+            String snippet = resolveWordForTheGivenCursorPosition(ctx, nodeFinder);
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    new SnippetBlock(":", snippet + ":", snippet + ":",
+                            ItemResolverConstants.VALUE_TYPE, SnippetBlock.Kind.VALUE)));
+            if (((ReFlagsOnOffNode) nodeAtCursor.parent().parent()).rhsReFlags().isEmpty()) {
+                // Eg: re `(?i<cursor>)`
+                completionItems.add(new SnippetCompletionItem(ctx,
+                        new SnippetBlock("-", snippet + "-", snippet + "-",
+                                ItemResolverConstants.VALUE_TYPE, SnippetBlock.Kind.VALUE)));
+            }
+            addReFlags(nodeAtCursor, ctx, completionItems, snippet);
+            return completionItems;
+        } else if (isCursorAfterReOnOffFlagMinusPosition(nodeAtCursor)) {
+            // Eg: re `(?i-<cursor>)`, re `(?-<cursor>)`
+            String snippet = resolveWordForTheGivenCursorPosition(ctx, nodeFinder);
+            addReFlags(nodeAtCursor, ctx, completionItems, snippet);
+            completionItems.add(new SnippetCompletionItem(ctx, RegexSnippet.DEF_COLON_CHAR.get()));
+            return completionItems;
+        } else if (nodeAtCursor.kind() == SyntaxKind.REGEX_TEMPLATE_EXPRESSION) {
+            // Eg: re `<cursor>`, re `<cursor>()`, re `<cursor>ABC`
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    Snippet.DEF_PARANTHESIS.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    Snippet.DEF_SQUARE_BRACKET.get()));
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    RegexpCompletionProvider.RegexSnippet.DEF_ASSERTION_START.get()));
+        }
+        if (nodeFinder.getTemplateExpressionNode() != null
+                && nodeFinder.getTemplateExpressionNode().textRange().endOffset() - 1
+                == ctx.getCursorPositionInTree()
+                && nodeAtCursor.kind() != SyntaxKind.RE_QUOTE_ESCAPE) {
+            // Eg: re `abc<cursor>`,
+
+            if (resolvedWord == null) {
+                resolvedWord = resolveWordForTheGivenCursorPosition(ctx, nodeFinder);
+            }
+            completionItems.add(new SnippetCompletionItem(ctx,
+                    new SnippetBlock("$", resolvedWord + "$", resolvedWord + "$",
+                            ItemResolverConstants.VALUE_TYPE, SnippetBlock.Kind.VALUE)));
+        }
```

Comment:
Let's move this block to a separate method and call this from the places where we don't have an early return. This will help us to reduce the if-else ladder. By doing this we can get rid of all the else statements and just use if statements. This will significantly improve the readability of the code. WDYT? @LakshanWeerasinghe

Reply:
yes, that's good. I'll do that.